internal/remoteconfig/remoteconfig.go

```diff
@@ -195,65 +193,77 @@ func newClient(config ClientConfig) (*Client, error) {
 }
 
 // Start starts the client's update poll loop in a fresh goroutine.
-// Noop if the client has already started.
+// If a client is already running, it will be stopped before starting a new one.
 func Start(config ClientConfig) error {
-	var err error
-	startOnce.Do(func() {
-		if !internal.BoolEnv("DD_REMOTE_CONFIGURATION_ENABLED", true) {
-			// Don't start polling if the feature is disabled explicitly
-			return
-		}
-		client, err = newClient(config)
-		if err != nil {
-			return
-		}
-		go func() {
-			ticker := time.NewTicker(client.PollInterval)
-			defer ticker.Stop()
-
-			for {
-				select {
-				case <-client.stop:
-					close(client.stop)
-					return
-				case <-ticker.C:
-					client.Lock()
-					client.updateState()
-					client.Unlock()
-				}
+	clientMu.Lock()
+	defer clientMu.Unlock()
+
+	if !internal.BoolEnv("DD_REMOTE_CONFIGURATION_ENABLED", true) {
+		// Don't start polling if the feature is disabled explicitly
+		return nil
+	}
+
+	// Stop existing client if running
+	if client != nil {
+		stopClientLocked()
+	}
```

Comment:
stopClientLocked() can be called unconditionally here, see stopClientLocked() for details.

```suggestion
	stopClientLocked()
```